### PR TITLE
Add Equation Protocol

### PR DESCRIPTION
**Pull request by DIO1009**
**NOTE**

> - If you would like to add a `volume` adapter please submit the PR [here](https://github.com/DefiLlama/adapters).
> - If you would like to add a `liquidations` adapter, please refer to [this readme document](https://github.com/DefiLlama/DefiLlama-Adapters/tree/main/liquidations) for details.

1. Once your adapter has been merged, it takes time to show on the UI. If more than 24 hours have passed, please let us know in Discord.
2. Please enable "Allow edits by maintainers" while putting up the PR.
3. Sorry, We no longer accept fetch adapter for new projects, we prefer the tvl to computed from blockchain data, if you have trouble with creating a the adapter, please hop onto our discord, we are happy to assist you.
4. Please fill the form below  **only if the PR is for listing a new protocol** else it can be ignored/replaced with reason/details about the PR
5. **For updating listing info** It is a different repo, you can find your listing in this file: https://github.com/DefiLlama/defillama-server/blob/master/defi/src/protocols/data2.ts, you can  edit it there and put up a PR
6. Do not edit/push `package-lock.json` file as part of your changes, we use lockfileVersion 2, and most use v1 and using that messes up our CI
7. No need to go to our discord and announce that you've created a PR, we monitor all PRs and will review it asap

---
##### Name (to be shown on DefiLlama):
Equation

##### Twitter Link:
https://twitter.com/EquationDAO

##### List of audit links if any:
https://github.com/EquationDAO/equation-contracts/blob/main/contracts/audits/ABDK_Equation_EquationDAO_v_1_0.pdf

##### Website Link:
https://equation.org/

##### Logo (High resolution, will be shown with rounded borders):
https://equation.trade/equation_logo.svg

##### Current TVL:
4.85 M

##### Treasury Addresses (if the protocol has treasury)
Protocol has no treasury

##### Chain:
Arbitrum only currently

##### Coingecko ID (so your TVL can appear on Coingecko, leave empty if not listed): (https://api.coingecko.com/api/v3/coins/list)
equation

##### Coinmarketcap ID (so your TVL can appear on Coinmarketcap, leave empty if not listed): (https://api.coinmarketcap.com/data-api/v3/map/all?listing_status=active,inactive,untracked&start=1&limit=10000)
None yet

##### Short Description (to be shown on DefiLlama):
Equation is a decentralized perpetual contract built on Arbitrum. With its innovative BRMM model, Equation provides both traders and Liquidity Providers (LPs) with up to 200x leverage, enabling traders to establish larger and unrestricted positions while enhancing capital efficiency for LPs. It offers the industry's lowest maintenance margin rate at just 0.25%, thus significantly reducing liquidation risk.

##### Token address and ticker if any:
Token address: 0x87AAfFdF26c6885f6010219208D5B161ec7609c0
Chain: Arbitrum
Ticker: EQU

##### Category (full list at https://defillama.com/categories) *Please choose only one:
Derivatives

##### Oracle used (Chainlink/Band/API3/TWAP or any other that you are using):
Chainlink

##### forkedFrom (Does your project originate from another project):
Project not forked

##### methodology (what is being counted as tvl, how is tvl being calculated):
Tvl is obtained by calculating the collective value of USDT in all pools.

##### Github org/user (Optional, if your code is open source, we can track activity):
https://github.com/EquationDAO

### CODE REVIEW

**Comment by llamatester** [comment]
The adapter at projects/equation exports TVL: 
        
 
 
    arbitrum-staking          91.31 k
    arbitrum-pool2            33.59 k
    arbitrum                  4.88 M
    staking                   91.31 k
    pool2                     33.59 k
    
    total                    4.88 M

File: projects/equation/index.js [hunk 1]
```diff
@@ -0,0 +1,39 @@
+const { sumTokens2 } = require("../helper/unwrapLPs");
+const { getLogs } = require('../helper/cache/getLogs')
+
+
+const config = {
+    arbitrum: { factory: '0xA91680161fBCeA942e164B42445aD6130D01541F', stakingContract: '0x3C77EEB8eC4716a6389a522eD590FbbD261ABE8e', EQU: '0x87AAfFdF26c6885f6010219208D5B161ec7609c0',V3Pool:['0xaFF6b531F5a06c00A21e34FD7BCe9cf024ef45F0','0xc7f83c20ad7DA85661546FECC19C27cb22ED53E3','0x7671e24bBD8B1FD52A4d936C8170927Fa36BAE3A','0xC0cF0f380DdB44dBCAF19A86d094C8BBA3efa04a'] },
```
Comment (g1nt0ki) [inline projects/equation/index.js:6]: hi @DIO1009 , thanks for the PR, the code is mostly good except for the pool2 part, you can only count the v3 nfts locked in your smart contract towards your pools, also unsure what last 3 addresses are

**Comment by llamatester** [comment]
The adapter at projects/equation exports TVL: 
        
 
 
    arbitrum-staking          115.09 k
    arbitrum-pool2            61.72 k
    arbitrum                  4.91 M
    staking                   115.09 k
    pool2                     61.72 k
    
    total                    4.91 M

**Comment by llamatester** [comment]
The adapter at projects/equation exports TVL: 
        
 
 
    arbitrum-staking          115.09 k
    arbitrum-pool2            61.72 k
    arbitrum                  4.91 M
    staking                   115.09 k
    pool2                     61.72 k
    
    total                    4.91 M

**Comment by llamatester** [comment]
The adapter at projects/equation exports TVL: 
        
 
 
    arbitrum-staking          139.41 k
    arbitrum-pool2            67.64 k
    arbitrum                  4.93 M
    staking                   139.41 k
    pool2                     67.64 k
    
    total                    4.93 M

**Pull request merged**